dbignite/data_model.py

```diff
@@ -33,9 +33,69 @@ def listDatabases(self) -> Iterable[Database]:
     def update(self) -> None:
         ...
 
-class FhirBundles(DataModel):
-    def __init__(self, path: str):
-        self.path = path
+class FhirBundles():
+    #
+    # Only supporting path representations currently
+    #
+    def __init__(self, defaultResource = None, **args):
+        from pyspark.sql import SparkSession
+        self.spark = SparkSession.getActiveSession()
+        self.df = None #force lazy evaluation
+        if defaultResource is None:
+            self.defaultResource = self.asWholeTextfile
+        else:
+            self.defaultResource = defaultResource
+        self.args = args
+
+    #
+    # Return json bundles as a DF 
+    #
+    def loadEntries(self):
+        if self.df is None:
+            self.df = self.defaultResource(**self.args)
+        return self.df
+
+    #
+    # ... 
+    #
+    @staticmethod
+    @udf(ArrayType(StringType()))  ## TODO change to pandas_udf
+    def _entry_json_strings(value):
+        """
+        UDF takes raw text, returns the
+        parsed struct and raw JSON.
+        """
+        bundle_json = json.loads(value)
+        return [json.dumps(e) for e in bundle_json["entry"]]
+
+
+    #
+    # Read a fhir bundle reasource as a whole text file (1 resource per file)
+    # @param path of the json bundles
+    def asWholeTextfile(self, path):
+        return (
+            self.spark.read.text(path, wholetext=True)
+              .select(explode(FhirBundles._entry_json_strings("value")).alias("entry_json"))
+              .withColumn("entry", from_json("entry_json", schema=ENTRY_SCHEMA))
+        ).cache()
+
+    #
+    # Read a fhir bundle resource as an inline json value (1 resource per line)
+    # 
+    def asInlineJson(self, path):
+        raise NotImplementedError("TODO...")
+
+    #
+    # Read a fhir bundle resource as an inline json value (1 resource per line, static 1 entry per bundle)
+    # 
+    def asInlineJsonSingleton(self, path):
```

Comment:
same as above: snake_case

Reply:
of course "python" wants you to use snake_case. Will change this